### PR TITLE
Rename StateIcon component to StatusIcon

### PR DESCRIPTION
**Pull request by JoanaMMoreira**

### CODE REVIEW

**Review by pseys — APPROVED** [review]

File: packages/core/src/form-field/StatusIndicator.tsx [hunk 1]
```diff
@@ -1,7 +1,7 @@
 import { IconProps } from "@jpmorganchase/uitk-icons";
 import cx from "classnames";
 import { ElementType } from "react";
-import { StateIcon } from "../state-icon";
+import { StatusIcon } from "../status-icon";
 import { Tooltip, TooltipProps, useTooltip } from "../tooltip";
 
 export type StateIndicatorState = "error" | "info" | "warning" | "success";
```
Comment (jeanlaleuf) [inline packages/core/src/form-field/StatusIndicator.tsx:7]: Shouldn't this type be called StatusIndicatorState?

File: packages/core/src/form-field/StatusIndicator.tsx [hunk 2]
```diff
@@ -44,14 +44,14 @@ export const StatusIndicator = (props: StatusIndicatorProps) => {
           ...TooltipProps,
         })}
       />
-      <StateIcon
-        {...getTriggerProps<typeof StateIcon>({
-          state,
+      <StatusIcon
+        {...getTriggerProps<typeof StatusIcon>({
+          status: state,
```
Comment (jeanlaleuf) [inline packages/core/src/form-field/StatusIndicator.tsx:49]: Shouldn't we simply rename the prop to "status"?

File: packages/lab/src/dialog/internal/DialogContext.ts [hunk 3]
```diff
@@ -1,8 +1,8 @@
 import { createContext } from "react";
-import { ValidationState } from "@jpmorganchase/uitk-core";
+import { ValidationStatus } from "@jpmorganchase/uitk-core";
 
 export const DialogContext = createContext<{
-  state?: ValidationState;
+  state?: ValidationStatus;
```
Comment (jeanlaleuf) [inline packages/lab/src/dialog/internal/DialogContext.ts:5]: DialogContext state prop should also be renamed

**Review by jeanlaleuf — COMMENTED** [review]
I think we need to propagate the change from state to status more thoroughly.

**Comment by JoanaMMoreira** [comment]
> I think we need to propagate the change from state to status more thoroughly.

Makes sense, I'll raise another PR

**Pull request merged**